### DISCUSSION TITLE
Operational and Support Questions about Ardor3D

### DISCUSSION THREAD

**Discussion by PowerUser1234** [Renanse/Ardor3D#103] [General]
I am trying to research the best 3D Java Library for my use needs.  I have honed in on Ardor3D, and would like the following questions answered.  I have knowledge of how Java3D 1.7 works.  However, when comparing it to something like OSG, even though that is in a separate language, Java3D lacks additional features such as special effects.

I have been hoping to find something more advanced than JavaFX 3D, too.

1)  I believe I want something which uses a Node Graph or Tree to assemble, hold, manipulate and process 3D related phenomena.  How does Ardor3D work in this regard?

2) Does Ardor3D have a capabilities system, which must be used, (or capabilities "compiling"), or not?  How does Ardor3D work in relation to that regard?

3)  I will probably like file format support, definitely reading by maybe also writing, for different 3D graphics toolkits, particularly Blender3D.

4)  I would like a pre-movement/collision detection system for the outsides of 3D objects.  Does Ardor3D support this?

5)  I seek support for 3D visual effects, certainly at or very similar to the extent that OSG, for C++, does. What does Ardor3D offer here?

6)  Naturally it all needs to paint on a Java Swing JComponent or a JPanel, somehow.  This is of course the case?

7)  There is a total Javadoc documentation included for use of the entire library?

8)  I want a product that is debugged, complete, Free Open Source Software, with online Forum support for free, which can be used in a commercial context for free, that is a professional level library.  Is Ardor3D this?

9)  In Ardor3D's view, how does it compare to the other Java 3D libraries out there, for Professional 3 Dimensional graphics, as well as for Game or Modelling purposes?

**Comment by Renanse** [comment]
I think just looking at your list of questions, Ardor3D is likely disqualified from what you are seeking on item 8 by itself - we don't have forums aside from the basic one provided here on GitHub.

While I am proud of the engine and what it has achieved to date, and it is in active use in places like NASA, it may not be the easiest for a new project to pick up and there's likely better choices, especially outside of Java.
